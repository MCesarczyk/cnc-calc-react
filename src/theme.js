export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}