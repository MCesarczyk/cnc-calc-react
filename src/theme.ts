export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
