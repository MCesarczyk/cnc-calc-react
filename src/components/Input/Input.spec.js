import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import "jest-styled-components";
import { Input } from ".";

test("Input component rendering properly", () => {
  const input = renderer.create(
    <ThemeProvider theme={theme} >
      <Input />
    </ThemeProvider>
  );
  
  let tree = input.toJSON();
  
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveStyleRule('background-color', 'beige', {
    modifier: ':invalid'
  })

  expect(tree).toHaveStyleRule('background-color', '#ddd', {
    modifier: ':disabled'
  })
});