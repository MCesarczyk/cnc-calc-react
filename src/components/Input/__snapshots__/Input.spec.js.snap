// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component rendering properly 1`] = `
.c0 {
  min-width: 200px;
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

.c0:invalid {
  background-color: beige;
}

.c0:disabled {
  background-color: #ddd;
}

@media (max-width:480px) {
  .c0 {
    width: 100%;
  }
}

<input
  className="c0"
/>
`;
